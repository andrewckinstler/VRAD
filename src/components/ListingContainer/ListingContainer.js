import React from 'react';
import Listing from '../Listing/Listing'

const ListingContainer = (props) => {
  return  'hi'
  // props.areas.map(area => {
  //   console.log(listing);
  //   return <Listing
  //     key={listing.address}
  //     listing={listing}
  //   />
  // })
}





export default ListingContainer
