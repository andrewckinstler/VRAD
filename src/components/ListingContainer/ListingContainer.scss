section {
  color: #0C4C38;

}
