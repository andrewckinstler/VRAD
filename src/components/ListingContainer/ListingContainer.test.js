import React from 'react';
import { shallow } from 'enzyme';
import ListingContainer from './ListingContainer';

describe('ListingContainer', () => {
  let wrapper;
  let mockListing;

  beforeEach(() => {
    mockListing = { name: 'test name', listing_id: 5 };
    wrapper = shallow(<ListingContainer
      listings={[]}
      favorites={[mockListing]}
      addFavorite={jest.fn()}
      setCurrentListing={jest.fn()}
    />);
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should start off with a default state', () => {
    const expected = {
      isLoading: false,
      listings: [],
      error: '',
    };
    expect(wrapper.state()).toEqual(expected);
  });

  it('should determine which listings are favorites when determineFavoriteStatus is called', () => {
    wrapper.setState({ listings: [mockListing] });
    wrapper.instance().determineFavoriteStatus = jest.fn(() => {
      return { favorite: true, listing_id: 5, name: 'test name' };
    });

    wrapper.instance().determineFavoriteStatus();
    expect(wrapper.instance().determineFavoriteStatus).toHaveReturnedWith({ favorite: true, listing_id: 5, name: 'test name' });
  });

  it('should create listing cards when createCards is called', () => {
    wrapper.instance().determineFavoriteStatus = jest.fn();

    wrapper.setState({ listings: [mockListing] });
    wrapper.instance().createCards();
    expect(wrapper.instance().determineFavoriteStatus).toHaveBeenCalled();
  });
});
