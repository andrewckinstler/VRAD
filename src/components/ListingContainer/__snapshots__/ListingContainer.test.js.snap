// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingContainer should create listing cards when createCards is called 1`] = `
<Fragment>
  <section>
    <Listing
      key="5"
      listing={
        Object {
          "listing_id": 5,
          "name": "test name",
        }
      }
    />
  </section>
</Fragment>
`;
